src/Glpi/Controller/Form/SubmitAnswerController.php

```diff
@@ -96,8 +96,13 @@ private function saveSubmittedAnswers(
         $post = $request->request->all();
         $provider = new EndUserInputNameProvider();
 
-        $answers = $provider->getAnswers($post);
-        $files = $provider->getFiles($post, $answers);
+        $delegation = [
+            'users_id'          => $post['delegation_users_id'] ?? null,
+            'use_notification'  => $post['delegation_use_notification'] ?? null,
+            'alternative_email' => $post['delegation_alternative_email'] ?? null,
```

Comment:
To get specific parameters, it is better to use the dedicated `$request->request->getString()/getInt()/...` methods to get better type support.

`$request->request->all();` should be only used when you need the full post data or an array of values.